Towerfall/Assets/_Main/Scripts/Player/PlayerManager.cs

```diff
@@ -7,35 +8,61 @@
 
 namespace Towerfall.Managers
 {
-    public partial class PlayerManager : IPlayerControllerInput
+    public partial class PlayerManager : IPlayerControllerInput, IPlayerControllerEvent
     {
         [Inject] private IPlayerInput _playerInput;
         [Inject] private IPlayerProperties _playerProperties;
 
-        private Subject<float> _jumpSubject = new Subject<float>();
-        private Subject<Vector2> _moveSubject = new Subject<Vector2>();
-        
-        public IObservable<float> Jump => _jumpSubject.AsObservable();
-        public IObservable<Vector2> Move => _moveSubject.AsObservable();
+        private Subject<float> _jumpStartSubject = new Subject<float>();
+        private Subject<float> _moveSubject = new Subject<float>();
+        private Subject<Vector2> _dashStartSubject = new Subject<Vector2>();
+
+        private Subject<Unit> _dashEndSubject = new Subject<Unit>();
         
-        private void PassJumpEvent()
+        public IObservable<float> JumpStart => _jumpStartSubject.AsObservable();
+        public IObservable<float> Move => _moveSubject.AsObservable();
+        public IObservable<Vector2> DashStart => _dashStartSubject.AsObservable();
+
+        public IObservable<Unit> DashEnd => _dashEndSubject.AsObservable();
+
+
+        private void PassJumpStartEvent()
         {
-            _jumpSubject.OnNext(_playerProperties.JumpForceMagnitude);
+            _jumpStartSubject.OnNext(_playerProperties.JumpForceMagnitude);
         }
 
-        private void PassMoveEvent(Vector2 moveDirection)
+        private void PassMoveEvent(float moveDirection)
         {
-            moveDirection.x *= _playerProperties.MoveAccelerationRate;
+            moveDirection *= _playerProperties.MoveAccelerationRate;
             _moveSubject.OnNext(moveDirection);
```

Comment:
Move the operation to `_moveSubject.OnNext();` so we keep as less lines as possible, in this case is desired cause is very simple logic, if it gets messier then we can add more variables here.

Reply:
212ca7af1e15d2e6e2dce944c83b800268220451